test/sdf_test.cpp

```diff
@@ -115,3 +115,21 @@ TEST(SDF, Resize) {
   EXPECT_EQ(layers.Status(), Manifold::Error::NoError);
   EXPECT_EQ(layers.Genus(), -8);
 }
+
+TEST(SDF, SineSurface) {
+  Mesh surface(LevelSet(
+      [](glm::vec3 p) {
+        float mid = glm::sin(p.x) + glm::sin(p.y);
+        return (p.z > mid - 0.5 && p.z < mid + 0.5) ? 1 : 0;
+      },
+      {glm::vec3(-4 * glm::pi<float>()), glm::vec3(4 * glm::pi<float>())}, 1));
+  Manifold smoothed = Manifold::Smooth(surface).Refine(2);
+
+  EXPECT_EQ(smoothed.Status(), Manifold::Error::NoError);
+  EXPECT_EQ(smoothed.Genus(), -2);
```

Comment:
@pca006132 I'm a little surprised this is -2 maybe some small disconnected object somewhere, or tiny hole? The model looks like it ought to be 0. I'm not too worried about it for now, but might be worth looking into at some point.